libevmasm/SemanticInformation.cpp

```diff
@@ -263,13 +263,45 @@ bool SemanticInformation::invalidatesMemory(Instruction _instruction)
 	case Instruction::CALLCODE:
 	case Instruction::DELEGATECALL:
 	case Instruction::STATICCALL:
+		return yul::SideEffects::Write;
```

Comment:
Hm, I think it is a bad design to use definitions from the `yul` namespace in the `libevmasm` namespace as that creates circular dependency.

Reply:
Most of the functions in here are yul-specific, but not all of them. We can move most of them into libyul and in the worst case create another enum and a translation function.